Adds satoshi port cli passthrough to ArmoryDB

TierNolan opened the pull request:
This should fix the satoshi port issue.  Sorry about the unintentionally misleading log file.

This at least connects when I tested it with a non-standard port.

TierNolan commented:
Just to clarify on "how the port got passed before", the log I sent you with the bug report was unintentionally misleading.

I have an explanation [here](https://bitcointalk.org/index.php?topic=5071932.msg48168698#msg48168698).  

The command line included
    
    Spawning DB with command: ./ArmoryDB.exe --db-type="DB_FULL" --cookie --satoshi-datadir="F:\bitcoin_data_dir\blocks" --datadir="G:\armory_data" --dbdir="G:\armory_data/database_dir --debug --satoshi-port=8500"
    

It made it look like satoshi-port was passed but it is an illusion due to the quotes.  The actual dbdir parameter is set as follows.
    
    --dbdir="G:\armory_data/database_dir --debug --satoshi-port=8500"
    

This means that dbdir is set to everything.  It even created a directory with the name "database_dir --debug --satoshi-port=8500".

I am not sure what happened.  It may have been a windows bug or a typo when I started armory from the command line.

goatpig commented:
I've seen the thread, I'm still trying to understand why the quotes got mangled. Unless you set the dbdir to that explicit line (opening one quote on dbdir on closing it on satoshi-port), I need to understand why it turned out this bad in the first place.

TierNolan commented:
When generating the logs, I used a windows command line.  I think I probably typo-ed the command line.

I did a quick test and this command line is accepted by windows
    
    "C:\Program Files (x86)\Armory\ArmoryQt.exe" --datadir=G:\armory_data --dbdir="G:\armory_data/database_dir --debug --satoshi-port=8500
    

It doesn't give an error that there is no closing quotes.   It just assumes the quote goes to the end of the line.

When I was testing on windows using the command prompt, I just copy pasted the shortcut and then edited it.  I am not sure why I added the quotes after dbdir though, but I may have done it with a previous run.  Perhaps, I added quotes at both ends and later deleted the 2nd one.

Assuming that I just added one quotes, and then used last command and appended --debug to generate detailed logs.  This would give the following commands.

Default port
    
    "C:\Program Files (x86)\Armory\ArmoryQt.exe" --datadir=G:\armory_data --dbdir="G:\armory_data/database_dir --debug
    

Custom port
    
    "C:\Program Files (x86)\Armory\ArmoryQt.exe" --datadir=G:\armory_data --dbdir="G:\armory_data/database_dir --debug --satoshi-port=8500
    

Looking at the default port [log file](https://hastebin.com/ixafuniheq.py), it looks like it has the same problem.
    
    2018-11-13 23:02:32 (INFO) -- ArmoryUtils.pyc:1268 - Invoked: C:\Program Files (x86)\Armory\ArmoryQt.exe --datadir=G:\armory_data --dbdir=G:\armory_data/database_dir --debug
    
This is the start of the log and it looks like all quotes are stripped.

However, down further down, it has the wrong directory.
    
    2018-11-13 23:02:32 (INFO) -- ArmoryUtils.pyc:1294 -     armoryDBDir     : G:\armory_data/database_dir --debug
    
This is the same as with the custom port [log file](https://hastebin.com/bemodofeho.py).
    
    2018-11-14 01:34:27 (INFO) -- ArmoryUtils.pyc:1294 -     armoryDBDir     : G:\armory_data/database_dir --debug --satoshi-port=8500
    

I think the problem was caused by user error by me and windows not caring if quotes are closed.
